mealie/services/migrations/cookn.py

```diff
@@ -0,0 +1,409 @@
+
+        if current_step:
+            steps.append(" ".join(current_step))
+
+        return steps
+
+    def _process_recipe_document(
+        self, _recipe_row: dict[str, Any], db: DSVParser
+    ) -> dict[str, str | list[str] | list[RecipeIngredient]]:
+        """Parses recipe row from the Cook'n recipe table."""
+        recipe_data: dict[str, str | list[str] | list[RecipeIngredient]] = {}
+
+        # Select db values
+        _recipe_id = db.get_data(_recipe_row, "ID")
+        _recipe_desc_row = db.query_by_id("temp_recipe_desc", "ID", [_recipe_id])[0]
+        _chapter_id = db.get_data(_recipe_desc_row, "PARENT")
+        _chapter_row = db.query_by_id("temp_chapter_desc", "ID", [_chapter_id])[0]
+        _cookbook_id = db.get_data(_chapter_row, "PARENT")
+        _cookbook_row = db.query_by_id("temp_cookBook_desc", "ID", [_cookbook_id])[0]
+        _media_row = db.query_by_id("temp_media", "ENTITY_ID", [_recipe_id])[0]
+
+        # Parse general recipe info
+        cookbook = db.get_data(_cookbook_row, "TITLE")
+        chapter = db.get_data(_chapter_row, "TITLE")
+        name = db.get_data(_recipe_desc_row, "TITLE")
+        description = db.get_data(_recipe_desc_row, "DESCRIPTION")
+        serves = db.get_data(_recipe_row, "SERVES")
+        # yields = db.get_data(_recipe_row, "YIELD")
+        prep_time = int(db.get_data(_recipe_row, "PREPTIME"))
+        cook_time = int(db.get_data(_recipe_row, "COOKTIME"))
+
+        recipe_data["recipeCategory"] = [cookbook + " - " + chapter]
+        recipe_data["name"] = name
+        recipe_data["description"] = description
+        recipe_data["recipeYield"] = serves
+        recipe_data["prepTime"] = format_time(prep_time)
+        recipe_data["performTime"] = format_time(cook_time)
+        recipe_data["totalTime"] = format_time(prep_time + cook_time)
+
+        # Parse image file
+        image_path = self._parse_media(_cookbook_id, _chapter_id, _recipe_id, db)
+        if image_path is not None:
+            recipe_data["image"] = [image_path]
+
+        # Parse ingredients
+        recipe_data["ingredients"] = self._parse_ingredients(_recipe_id, db)
+
+        # Parse instructions
+        recipe_data["recipeInstructions"] = self._parse_instructions(db.get_data(_recipe_row, "INSTRUCTIONS"))
+
+        return recipe_data
+
+    def _process_cookbook(self, path: Path) -> None:
+        """Processes contents of a zip file."""
+        source_dir = self.get_zip_base_path(path)
+        db = DSVParser(source_dir)
+        # Load units and foods from Cook'n
+        self._parse_units_table(db)
+        self._parse_foods_table(db)
+        # Reload DataMatcher with updated tables
+        self.matcher = DataMatcher(self.db, food_fuzzy_match_threshold=95, unit_fuzzy_match_threshold=100)
+
+        # Load recipes from cookn
+        _recipe_table = db.get_table("temp_recipe")
+        recipes_as_dicts = [self._process_recipe_document(_recipe_row, db) for _recipe_row in _recipe_table]
```

Comment:
Could we loop over these and skip any with errors? That way if there's one recipe with bad data we don't abort the entire import. See the Tandoor migration for example:
```python
for i, recipe_zip_file in ...:
    try:
        recipe_json_path = ...
        with open(recipe_json_path) as f:
            recipes_as_dicts.append(self._process_recipe_document(recipe_source_dir, json.load(f)))

    except Exception as e:
        self.report_entries.append(
            ReportEntryCreate(
                report_id=self.report_id,
                success=False,
                message="Failed to parse recipe",
                exception=f"{type(e).__name__}: {e}",
            )
        )
```